packages/material-ui/src/Dialog/Dialog.spec.tsx

```diff
@@ -0,0 +1,6 @@
+import * as React from 'react';
+import { Dialog } from '@material-ui/core';
+
+function optionalChildrenTest() {
+  <Dialog open />;
```

Comment:
Without the change in this PR I get the following error:

```
yarn workspace @material-ui/core typescript
yarn workspace v1.22.4
yarn run v1.22.4
$ tslint -p tsconfig.json "{src,test}/**/*.{ts,tsx}" && tsc -p tsconfig.test.json

ERROR: /Users/minhnguyen/Development/mui-org/material-ui/packages/material-ui/src/Dialog/Dialog.spec.tsx:5:4 - TypeScript@next compile error: 
Property 'children' is missing in type '{ open: true; }' but required in type 'DialogProps'.

error Command failed with exit code 2.
```

Reply:
@eps1lon can also change this to say `open={false}` if you think that would be a more representative scenario.